cosmo_tester/test_suites/broker_management/conftest.py

```diff
@@ -0,0 +1,112 @@
+import os
+import time
+
+import pytest
+
+from cosmo_tester.framework.test_hosts import BootstrapBasedCloudifyManagers
+from cosmo_tester.framework import util
+
+
+@pytest.fixture()
+def brokers(cfy, ssh_key, module_tmpdir, attributes, logger):
+    for _brokers in _get_brokers(cfy, ssh_key, module_tmpdir,
+                                 attributes, logger):
+        yield _brokers
+
+
+@pytest.fixture()
+def broker(cfy, ssh_key, module_tmpdir, attributes, logger):
+    for _brokers in _get_brokers(cfy, ssh_key, module_tmpdir, attributes,
+                                 logger, instance_count=1):
+        yield _brokers[0]
+
+
+class JustHosts(BootstrapBasedCloudifyManagers):
+    def _bootstrap_managers(self):
+        pass
+
+
+def skip(*args, **kwargs):
+    return True
+
+
+def _get_brokers(cfy, ssh_key, module_tmpdir, attributes, logger,
+                 instance_count=3):
+    hosts = JustHosts(
+        cfy, ssh_key, module_tmpdir, attributes, logger,
+        number_of_instances=instance_count,
+    )
+
+    tempdir = hosts._tmpdir
+
+    cert_paths = {
+        'ca': {
+            'base': os.path.join(tempdir, 'ca.'),
+        },
+    }
+    for node in range(instance_count):
+        node = 'node' + str(node + 1)
+        cert_paths[node] = {
+            'base': os.path.join(tempdir, node + '.'),
+        }
+    for paths in cert_paths.values():
+        paths['cert'] = paths['base'] + 'crt'
+        paths['key'] = paths['base'] + 'key'
+    util.generate_ca_cert(cert_paths['ca']['cert'], cert_paths['ca']['key'])
+
+    try:
+        for node in hosts.instances:
+            node.verify_services_are_running = skip
+            node.upload_necessary_files = skip
+            node.upload_plugin = skip
+            node.enter_sanity_mode = skip
+
+        hosts.create()
+
+        for node_num, node in enumerate(hosts.instances, 1):
+            node_id = 'node' + str(node_num)
+            with node.ssh() as fabric_ssh:
+                node.hostname = str(fabric_ssh.run('hostname -s'))
+                logger.info('Preparing {}'.format(node.hostname))
+            util.generate_ssl_certificate(
+                [node.hostname, node.private_ip_address],
+                node.hostname,
+                cert_paths[node_id]['cert'],
+                cert_paths[node_id]['key'],
+                cert_paths['ca']['cert'],
+                cert_paths['ca']['key'],
+            )
+
+            node.put_remote_file(
+                local_path=cert_paths[node_id]['cert'],
+                remote_path='/tmp/rabbit.crt',
+            )
+            node.put_remote_file(
+                local_path=cert_paths[node_id]['key'],
+                remote_path='/tmp/rabbit.key',
+            )
+            node.put_remote_file(
+                local_path=cert_paths['ca']['cert'],
+                remote_path='/tmp/rabbit.ca',
+            )
+
+            node.additional_install_config = {
+                'rabbitmq': {
```

Comment:
the two brokers aren't in a cluster?

Reply:
They don't start clustered- if they did, some of the tests would require removing and reinstalling.